pkg/binding/bindingpolicy-resolution.go

```diff
@@ -48,21 +49,41 @@ type bindingPolicyResolution struct {
 	requiresSingletonReportedState bool
 }
 
-// ensureObjectIdentifierWithVersion ensures that an object identifier exists
-// in the resolution and is associated with the given resource version.
+// objectData stores the resource version and status collectors for an object.
+type objectData struct {
+	ResourceVersion  string
+	StatusCollectors sets.Set[string]
+}
+
+// ensureObjectIdentifier ensures that an object identifier exists
```

Comment:
FYI, I think that the name of this method would be clearer if it talked about more than the identifier for the object. Something like `ensureObjectData` might be better.